File: cpp-examples/composition/Section.cpp
#include "Section.h"
#include <iostream>

Section::Section(const std::string& name) : name(name) {}

Section::~Section() {
    for (auto book : books) {
        delete book;
    }
    books.clear();
}

void Section::addBook(Book* book) {
    books.push_back(book);
}

void Section::printDetails() const {
    std::cout << "Section: " << name << std::endl;
    for (const auto& book : books) {
        book->printDetails();
    }
}

File: python-examples/composition/Section.py
class Section:
    def __init__(self, name):
        self.name = name
        self.books = []

    def add_book(self, book):
        self.books.append(book)

    def print_details(self):
        print(f"Section: {self.name}")
        for book in self.books:
            book.print_details()
